tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -2682,11 +2685,16 @@ def spatial_softmax(features,
       features = array_ops.reshape(
           array_ops.transpose(features, [0, 3, 1, 2]), [-1, height * width])
 
-    softmax_attention = nn.softmax(features/temperature)
+    if activation_fn is not None:
+      attention = activation_fn(features/temperature)
+    else:
+      attention = features/temperature
+
+    attention = nn.softmax(features/temperature)
     expected_x = math_ops.reduce_sum(
```

Comment:
I'm not sure if any activation function would make sense here. The way of computing expected_x and expected_y assumes that the attention mask sum 1.
You would need to show that using other activation functions make sense and can compute reasonable x, y.

Reply:
whoops I forgot to delete the nn.softmax line, since the `activation_fn` line would call softmax by default. The main other activation function that makes sense here is `sigmoid` if you have a 1 class problem. I chose to make it a function parameter because it is in line with other calls in this file.